server/webapp/WEB-INF/rails.new/config/routes.rb

```diff
@@ -228,6 +228,9 @@
         patch :update, on: :member
       end
 
+      get 'stages/:pipeline_name/:pipeline_counter/:stage_name/:stage_counter/instance' => 'stages#show', constraints: {pipeline_name: PIPELINE_NAME_FORMAT, pipeline_counter: PIPELINE_COUNTER_FORMAT, stage_name: STAGE_NAME_FORMAT, stage_counter: STAGE_COUNTER_FORMAT}, as: :stage_instance_by_counter_api
+      get 'stages/:pipeline_name/:stage_name/history' => 'stages#history', constraints: {pipeline_name: PIPELINE_NAME_FORMAT, stage_name: STAGE_NAME_FORMAT}, as: :stage_history_api
+
```

Comment:
I don't think so. It'll break something for sure. If we know we're going to break something, we should put some effort towards finding out why it was the way it was.